src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -82,25 +106,11 @@ private[xml] object StaxXmlParser extends Serializable {
     }
 
     xml.mapPartitions { iter =>
-      val factory = XMLInputFactory.newInstance()
-      factory.setProperty(XMLInputFactory.IS_NAMESPACE_AWARE, false)
-      factory.setProperty(XMLInputFactory.IS_COALESCING, true)
-      val filter = new EventFilter {
-        override def accept(event: XMLEvent): Boolean =
-          // Ignore comments. This library does not treat comments.
-          event.getEventType != XMLStreamConstants.COMMENT
-      }
 
       iter.flatMap { xml =>
-        // It does not have to skip for white space, since `XmlInputFormat`
-        // always finds the root tag without a heading space.
-        val eventReader = factory.createXMLEventReader(new StringReader(xml))
-        val parser = factory.createFilteredReader(eventReader, filter)
+        val parser = filteredReader(xml)
```

Comment:
This will called for every record. We should make the factory for each partition like the previous code, and make each reader for each record. Such pattern of computation per record causes a considerable performance regression.

Reply:
@HyukjinKwon I think I've found a semi-simple solution in the latest commit. Look good?